packages/ember-template-compiler/lib/index.js

```diff
@@ -31,13 +30,8 @@ registerPlugin('ast', TransformInputOnToOnEvent);
 registerPlugin('ast', TransformTopLevelComponents);
 registerPlugin('ast', TransformUnescapedInlineLinkTo);
 registerPlugin('ast', AssertNoEachIn);
-
-if (_Ember.ENV._ENABLE_LEGACY_VIEW_SUPPORT) {
-  registerPlugin('ast', TransformEachIntoCollection);
-} else {
-  registerPlugin('ast', AssertNoViewAndControllerPaths);
-  registerPlugin('ast', AssertNoViewHelper);
-}
+registerPlugin('ast', AssertNoViewAndControllerPaths);
```

Comment:
These should only be registered if the legacy view flag is off. We definitely should have a follow up PR to remove the flag and all usages of it, but it should be a separate change from this one IMO.